automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/WorkspaceDataPage.scala

```diff
@@ -18,14 +22,49 @@ class WorkspaceDataPage(namespace: String, name: String)(implicit webDriver: Web
 
   val dataTable = Table("entity-table")
   private val importMetadataButton = Button("import-metadata-button")
+  private val downloadMetadataButton = Button("download-metadata-button")
 
-  def importFile(file: String): Unit = {
+  def importFile(filePath: String): Unit = {
     importMetadataButton.doClick()
     val importModal = await ready new ImportMetadataModal
-    importModal.importFile(file)
+    importModal.importFile(filePath)
     dataTable.awaitReady()
   }
 
+  def importFile(file: File): Unit = {
+    importFile(file.getAbsolutePath)
+  }
+
+  /**
+    * Downloads the metadata currently being viewed. If downloadPath is given, the file is given a
+    * timestamped name and moved from that location into the "downloads" directory off the current
+    * working directory.
+    *
+    * @param downloadPath the directory where the browser saves downloaded files
+    * @return the relative path to the moved download file, or None if downloadPath was not given
+    */
+  def downloadMetadata(downloadPath: Option[String] = None): Option[String] = {
+    downloadMetadataButton.doClick()
+    /*
+     * Downloading a file will open another window while the download is in progress and
+     * automatically close it when the download is complete.
+     */
+    await condition { windowHandles.size == 1 }
+
+    for {
+      path <- downloadPath
+      entityType <- find(CssSelectorQuery(downloadMetadataButton.element.queryString)).get.attribute("data-entity-type")
+    } yield archiveDownloadedFile(s"$path/$entityType.txt")
+
+    def archiveDownloadedFile(sourcePath: String): String = {
```

Comment:
may want to add comments in here to explain clearly why/what - to be helpful to future you/devs